### DISCUSSION TITLE
Questions about QStateMachine

### DISCUSSION THREAD

**Discussion by mikeriedel** [miurahr/aqtinstall#542] [Q&A] [answered]
Hi, I just discovered this python project and realized it is the solution to a problem I've had for years. I'm currently working on setting up a build for a Qt project that uses the "Qt State Machines" module. I'm not seeing it in the modules list. Am I missing something? Is there a reason its missing?

-----Terminal output------
python -m aqt list-qt windows desktop --modules 6.2.4 win64_msvc2019_64
debug_info qt3d qt5compat qtactiveqt qtcharts qtconnectivity qtdatavis3d qtimageformats qtlottie qtmultimedia qtnetworkauth qtpositioning qtquick3d qtquicktimeline qtremoteobjects qtscxml qtsensors qtserialbus qtserialport qtshadertools qtvirtualkeyboard qtwebchannel qtwebengine qtwebsockets qtwebview

python -m aqt list-qt windows desktop --modules 6.3.1 win64_msvc2019_64
debug_info qt3d qt5compat qtactiveqt qtcharts qtconnectivity qtdatavis3d qtimageformats qtlanguageserver qtlottie qtmultimedia qtnetworkauth qtpdf qtpositioning qtquick3d qtquicktimeline qtremoteobjects qtscxml qtsensors qtserialbus qtserialport qtshadertools qtvirtualkeyboard qtwebchannel qtwebengine qtwebsockets qtwebview

python -m aqt list-qt windows desktop --modules 6.4.0 win64_msvc2019_64
debug_info qt3d qt5compat qtactiveqt qtcharts qtconnectivity qtdatavis3d qthttpserver qtimageformats qtlanguageserver qtlottie qtmultimedia qtnetworkauth qtpdf qtpositioning qtquick3d qtquick3dphysics qtquicktimeline qtremoteobjects qtscxml qtsensors qtserialbus qtserialport qtshadertools qtspeech qtvirtualkeyboard qtwebchannel qtwebengine qtwebsockets qtwebview

**Comment by ddalcino** [comment, marked as answer]
Have you tried using qtscxml? The Updates.xml file describes that module as “Qt state machine”, so it might be what you’re looking for.

aqt list-qt doesn’t have a function that prints the descriptions of modules, the way aqt list-tool does. Maybe that would have helped here.

**Reply by mikeriedel** [reply]
I see what you're saying now, the talk about aqt list-tool threw me off. The naming is super surprising considering what I previously typed however that looks promising. I accidentally rebased over my build script so its going to take me some time to test this.

---------------Old Text---------------
I'm pretty sure they're different. I'll give it a try.  I think SCXML is a parser of xml statemachines and what I'm looking for is the underlying classes that allow a Qt/C++ statemachine to compile. 

> The Qt SCXML module differs from the [State Machine framework](https://doc.qt.io/qt-6/qtstatemachine-overview.html) in the Qt [StateMachine](https://doc.qt.io/qt-6/qml-qtqml-statemachine-statemachine.html) module in that Qt SCXML provides a conforming processor that can parse and process conforming SCXML documents.

https://doc.qt.io/qt-6/qtscxml-overview.html

![image](https://user-images.githubusercontent.com/15795274/181341617-11baaa1e-37ae-4294-b9d3-025c99edbc4d.png)

**Reply by miurahr** [reply]
There is source code of ``SCXML`` https://download.qt.io/archive/qt/6.3/6.3.1/submodules/qtscxml-everywhere-src-6.3.1.zip
You can find a source ``src/statemachine`` and ``src/statemachineqml`` in it.
Is it what you look for?

![image](https://user-images.githubusercontent.com/123720/181388288-3ea01352-67cb-4bfd-afd7-334b12df2a4a.png)

**Reply by ddalcino** [reply]
Take a look at this entry in [the Updates.xml file](https://download.qt.io/online/qtsdkrepository/mac_x64/desktop/qt6_631/Updates.xml) here:
<img width="876" alt="Screen Shot 2022-07-27 at 5 53 09 PM" src="https://user-images.githubusercontent.com/6743441/181397285-7408ded3-2658-43f2-b28a-f746502d00c3.png">

To install this module with aqt you have to call it `qtscxml`. aqt is getting this name from the "Name" element in the xml file. The "DisplayName" element is "Qt State Machine". The graphical Qt installer reads the "DisplayName" property from this same xml file when it's drawing the names of the modules in your screenshot.

According to this xml file, if you want to install the "Qt State Machine" module, the only way to do it with aqt is to call it `qtscxml`.

**Reply by mikeriedel** [reply]
Sorry for the slow reply. You guys are right, {Name: "qtscxml"; DisplayName: "Qt State Machines"} was what I was looking for!!! Thanks so much for the help.
